### ISSUE TITLE
Update static theme preview image sizes proposal/discussion

### ISSUE THREAD

**Issue by rebmullin** [mozilla/addons-server#8412]
Static theme preview image sizes:
680x92 - "full" - keep this for header on detail page 

Note: we're currently using 680x92 size on the search listing view since it has comparable height to lwt. These image's width will be stretched a bit at the larger bps (which is what we were currently doing) until the lwt have these same sizes and/or are migrated.

*proposed change (from 670x64)*:
760x64 - "thumb" -  760px is more ideal than 670px/680px size for the search listing view so nothing needs to be stretched. We'll start using this when lwt has been migrated over and/or has these same sizes (so there is no height discrepancies).

@pwalm are you okay with this? Am I missing anything, are there other views to consider?

// cc @eviljeff

**Comment by eviljeff** [comment]
I'm wary about doing this until mozilla/addons-frontend#5174 is resolved (either as wontfix or otherwise).

**Comment by pwalm** [comment]
@rebmullin yep, fine by me. Do you need new assets?

**Comment by eviljeff** [comment]
@rebmullin if I'm remembering the outcome of the meeting in SF correctly, we are going ahead with your size changes in the issue description and the thumbnail will be used for the search results only.  

The shelves (i.e. the boxes that currently contain 4 short images) will change to use the main preview size, but scaled down so they're not cropped (and display 3 on a line).  Have I got all that correct?

**Comment by rebmullin** [comment]
@eviljeff  Actually I am not sure about the 670x64 (suggested thumbnail change above) anymore b/c if we scale this down (for mobile/shelves), I think the height *could* get too short:

comparing 1 breakpoint example:
scaling down 680x92
![Alt text](https://monosnap.com/image/s4b5cDAljUbWEyCRq0BhJJvbTjgE9w.png)

scaling down 670x64
![Alt text](https://monosnap.com/image/ep7ifSCuSra9GAwiKmBwgaOQfGfoMx.png)

maybe this is okay..? 

Right now we are using the full version (680x92) so I'm thinking this might be a little better if we are going to scale down..unless @pwalm really wants to keep the 64px height for search listing desktop...

However we also talked about having a 3rd/mobile size...is that possible? Something with the same aspect ratio - something like ~466x64..? If we can get this additional size, then I think we can use this for the shelves/grid cols and even search listing on mobile.

So basically this is what I was thinking now:
-Header size (size: don't change/same as noted above)
-Search listing size (size: ~ 670x92 - for search listing desktop)
-mobile / shelves/grids size (size: ~ 466x64 - for search listing mobile and for shelves/grids cols, see on homepage and theme author cards)

(I also heard a brief discussion on having additional sizes for different resolutions but I am not sure what the verdict was on that)

// cc @pwalm

**Comment by pwalm** [comment]
Can we scale horizontally without changing the height?

**Comment by rebmullin** [comment]
hey @pwalm This is actually what we are doing right now - scaling the width with a fixed height (which causes cropping).  We need to be able to keep the same aspect ratio - which means scaling height too - in order for this to work.  Even if we enforce a small min height I think with the different alignment issues, we will run into issues...

Im’ just thinking tho - maybe we could increase the heights a bit on each original size while keeping the same approx widths..? Then as it scales down, we have a little bit more height to work with. 

Is your concern that on tablet/ mobile this will be too small/short?

**Comment by eviljeff** [comment]
Just to be clear, what you're asking for is:

> -Header size 680x92
> -Search listing size 670x92
> -mobile / shelves/grids size 466x64

Do you really need both "Header size" and "Search listing size" when they're 10px width different?

**Comment by rebmullin** [comment]
ahh mistype! So sorry for confusion but search listing should be 760x92 (for reason discussed above about ideally not stretching images) and mobile ideally should have the same aspect ratio...so maybe about ~528x64. 

height is still TBD tho (per my last comment) so this still might change a bit..

**Comment by eviljeff** [comment]
There are some easy-ish options here:
 - if the mobile size is the same aspect ratio as the search listing size then just use search listing size for mobile/shelves size and downscale with css.  So `preview[0].image_url` is header size ; `preview[0].thumbnail_url` is search listing size.
- have a second preview - i.e. one preview per aspect ratio - so each would have a full size and a thumbnail size.  So `preview[0].image_url` is header size; `preview[1].image_url` is search listing size; `preview[1].thumbnail_url` is mobile/shelves size.

In the former you'd get:
-Header size 680x92
-Search listing size 760x92

In the latter you'd get:
- Header size 680x92 (and ~466x64 or some other size at the same ratio)
- Search listing size 760x92 and 528x64

Or just drop one of the aspect ratios, as you suggested, and only have 760x92 size.

Tbh, having one preview per aspect ratio would fit more with the idea of having other resolutions in the future.

**Comment by pwalm** [comment]
@rebmullin @eviljeff The latter sounds cleaner with these sizes/ratios:
- Header is 680x92
- Listing is 760x92 / 528x64

Will this solve the issue? If so, I'll provide updated images (with the reduced prominence of icons/address bar text)

**Comment by rebmullin** [comment]
@eviljeff @pwalm, It sounds like we all agree the latter/2nd option would be the best solution since it provides a mobile version and we can maybe more easily add other resolutions in the near future. 

so @pwalm, I think we will have 1 more "mobilish" size -  so something like this for final sizes:

Header:  680x92 / 473/64
Listing: 760x92 / 528x64

does this look good to everyone?

**Comment by pwalm** [comment]
Fantastic. I'll whip up those new sizes this afternoon and attach them to this issue.

**Comment by pwalm** [comment]
Attached! 
[amo static theme svgs.zip](https://github.com/mozilla/addons-server/files/2121385/amo.static.theme.svgs.zip)

**Comment by eviljeff** [comment]
@rebmullin just new ones (a new version will be sufficient).  If it's a big problem we can just delete the old submissions from dev.

**Comment by rebmullin** [comment]
ok sounds good, thanks

**Comment by eviljeff** [comment]
QA:
- Submit a static theme either with an .xpi or via the wizard as a listed version.
- Trivially check the preview was rendered in the last step of the upload flow.  Should be the same size as before, 680x92, but with slightly smaller circles to represent icons (and a few other minor tweaks - see header_large.svg in pwalm's zip)
- to check other sizes easiest way is to use the API, as I'm not sure what frontend exposes currently.
  - preview 0 (the first preview) is the header preview, so should be 680x92 for image and 473x64 for thumbnail
  - preview 1 (the second preview) is the list preview, so should be 760x92 for image and 528x64 for thumbnail
  - the images referenced in urls should be the correct sizes and match the other svgs in pwalm's zip)

**Comment by AlexandraMoga** [comment]
I've tested the new ST previews based on the above recommendations and here are my results:
- the new preview styles use smaller circles and thinner lines to represent text, icons and other browser elements 

![image](https://user-images.githubusercontent.com/31961530/42088064-f901a892-7ba0-11e8-9f6e-2a3bb9a16f22.png)


- the API exposes the new preview sizes (image and thumbnail), with two previews in total. I've noticed that the second thumbnail size is 529x64 instead of 528x64, but I think that's negligible (in the specs this preview was set to 528.696x64 anyway)
  example: `https://addons-dev.allizom.org/api/v3/addons/addon/iarba/` 

![image](https://user-images.githubusercontent.com/31961530/42087732-c136349c-7b9f-11e8-811c-388e59173d63.png)

- the image URL's are also exposing the correct size and elements highlighted in the specs

@eviljeff 

- there is one issue for DevHub and RevTools preview where two previews are being generated now. I filed  #8693 as a follow up
